### PR TITLE
RD-2396: Replacing 's3' gem with 'aws-sdk-s3'

### PR DESCRIPTION
**Pull request by yashasvj**
To address AWS::S3::PermanentRedirect error when trying to access a non default
region - bucket.

Ref: https://stackoverflow.com/questions/26533245/the-authorization-mechanism-you-have-provided-is-not-supported-please-use-aws4?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

### CODE REVIEW

File: lib/content_caching/adapters/aws.rb [hunk 1]
```diff
@@ -33,17 +33,23 @@ def delete document_path
...
 
       def service
-        @service ||= S3::Service.new(
-          :access_key_id => self.options[:aws_access_key_id],
-          :secret_access_key => self.options[:aws_secret_access_key],
-          :use_ssl => true,
-          :use_vhost => true
+        @service ||= ::Aws::S3::Resource.new(
+          credentials: aws_credentials,
+          region: self.options[:region],
```
Comment (thomas-riboulet-finalcad) [inline lib/content_caching/adapters/aws.rb]: you pass region in 2 places, is that normal ?

File: lib/content_caching/adapters/aws.rb [hunk 2]
```diff
@@ -33,17 +33,23 @@ def delete document_path
...
+          use_ssl: true,
```
Comment (thomas-riboulet-finalcad) [inline lib/content_caching/adapters/aws.rb]: are those 2 options also supported by Aws::S3::Resource ?

File: lib/content_caching/adapters/aws.rb (hunk 1 above)
Comment (yashasvj) [reply lib/content_caching/adapters/aws.rb]: Oops. That was by mistake. Will correct it.

File: lib/content_caching/adapters/aws.rb [hunk 3]
```diff
@@ -33,17 +33,20 @@ def delete document_path
...
+          region: self.options[:region]
         )
       end
 
       def bucket
         @bucket ||= service.bucket(self.options[:directory])
       end
+
+      def aws_credentials
+        ::Aws::Credentials.new self.options[:aws_access_key_id],
```
Comment (thomas-riboulet-finalcad) [inline lib/content_caching/adapters/aws.rb]: please use () , we are using there everywhere, let's keep it that way

Comment (yashasvj) [reply lib/content_caching/adapters/aws.rb]: Noted. Thanks.

**Pull request merged**